lms/djangoapps/support/views/course_entitlements.py

```diff
@@ -0,0 +1,144 @@
+"""
+Support tool for changing and granting course entitlements
+"""
+from django.contrib.auth.models import User
+from django.db.models import Q
+from django.http import HttpResponseBadRequest
+from django.utils.decorators import method_decorator
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions, status, viewsets
+from rest_framework.response import Response
+from six import text_type
+
+from entitlements.api.v1.permissions import IsAdminOrAuthenticatedReadOnly
+from entitlements.api.v1.serializers import SupportCourseEntitlementSerializer
+from entitlements.models import CourseEntitlement, CourseEntitlementSupportDetail
+from openedx.core.djangoapps.cors_csrf.authentication import SessionAuthenticationCrossDomainCsrf
+from lms.djangoapps.support.decorators import require_support_permission
+
+
+class EntitlementSupportView(viewsets.ModelViewSet):
+    """
+    Allows viewing and changing learner course entitlements, used the support team.
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthenticationCrossDomainCsrf,)
+    permission_classes = (permissions.IsAuthenticated, IsAdminOrAuthenticatedReadOnly,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = SupportCourseEntitlementSerializer
+
+    def filter_queryset(self, queryset):
+        try:
+            username_or_email = self.request.GET.get('user')
```

Comment:
We should be explicit in our usage of parameters.  This should be split into user and email.  Using user for both is misleading.

Reply:
I think keeping it in just the one variable is useful so that support team members can specify either the username or the email of the user for whom they are searching

Reply:
It is unclear to the API user though. Is this the pattern in the other APIs?

Reply:
This is the pattern used by course enrollments.